sqlglot/optimizer/annotate_types.py

```diff
@@ -523,7 +525,7 @@ def _annotate_with_type(
         self, expression: E, target_type: exp.DataType | exp.DataType.Type
     ) -> E:
         self._set_type(expression, target_type)
-        return self._annotate_args(expression)
+        return expression
```

Comment:
We don't need this anymore, let's just make `_set_type` return its input so that it's composable.